libs/partners/mongodb/tests/integration_tests/test_vectorstores.py

```diff
@@ -34,17 +39,50 @@ def _insert_texts(self, texts: List[str], metadatas: List[Dict[str, Any]]) -> Li
             timeout -= INTERVAL
         return ids
 
+    def create_index(
+        self,
+        dimensions: int,
+        filters: Optional[List[Dict[str, str]]] = None,
+        update: bool = False,
+    ):
+        result = super().create_index(
+            dimensions=dimensions, filters=filters, update=update
+        )
+        timeout = TIMEOUT
+        index_ready = False
+        while not index_ready and timeout >= 0:
+            if indexes := list(
+                self._collection.list_search_indexes(name=self._index_name)
+            ):
+                if indexes[0].get("status") == "READY":
+                    index_ready = True
```

Comment:
Suggest returning early here to avoid a potential redundant list_search_indexes later on.